### PR TITLE
support api 2.0

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: response.js [hunk 1]
```diff
@@ -21,6 +21,13 @@ const normalizeHeaders = (req, options) => {
       // do nothing
     }
   }
+  if (req.http && req.http.method) {
```
Comment (jgallen23) [inline response.js]: double check this, I think it's req.requestContext

File: test/test.response.js [hunk 2]
```diff
@@ -77,3 +77,21 @@ tap.test('response method ', async t => {
...
 });
+
+tap.test('response method', async t => {
+  const { reply, response } = require('../');
+  const handler = response((req) => {
+    t.equal(req.path, '/api');
+    t.equal(req.method, 'post');
+    return reply.html('yay');
+  });
+  const res = await handler({
```
Comment (jgallen23) [inline test/test.response.js]: can you drop in the full example request from the documentation?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: response.js (hunk 1 above)
Comment (orthagonal) [reply response.js]: Are we sure arc will present it in this form for us?  req.requestContext is also a thing in 1.0 but doesn't have the 'http' object.

File: response.js [hunk 3]
```diff
@@ -69,6 +76,7 @@ module.exports = function(requestHandler, passedOptions = {}) {
   // default is true:
   const redirectTrailingSlash = options.redirectTrailingSlash === undefined ? true : options.redirectTrailingSlash;
   return async function(req) {
+    normalizeHeaders(req, options);
     const method = req.httpMethod || req.method;
```
Comment (jgallen23) [inline response.js]: don't need this anymore

File: response.js [hunk 4]
```diff
@@ -69,6 +76,7 @@ module.exports = function(requestHandler, passedOptions = {}) {
   // default is true:
   const redirectTrailingSlash = options.redirectTrailingSlash === undefined ? true : options.redirectTrailingSlash;
   return async function(req) {
+    normalizeHeaders(req, options);
```
Comment (jgallen23) [inline response.js]: lets rename this to just normalize

**Review by jgallen23 — CHANGES_REQUESTED** [review]

**Review by jgallen23 — APPROVED** [review]

**Pull request merged**